homeassistant/components/nfandroidtv/notify.py

```diff
@@ -14,115 +13,69 @@
     BaseNotificationService,
 )
-from homeassistant.const import CONF_HOST, CONF_TIMEOUT, HTTP_OK, PERCENTAGE
+from homeassistant.const import ATTR_ICON, CONF_HOST, CONF_TIMEOUT
+from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
 
-_LOGGER = logging.getLogger(__name__)
-
-CONF_DURATION = "duration"
-CONF_FONTSIZE = "fontsize"
-CONF_POSITION = "position"
-CONF_TRANSPARENCY = "transparency"
-CONF_COLOR = "color"
-CONF_INTERRUPT = "interrupt"
-
-DEFAULT_DURATION = 5
-DEFAULT_FONTSIZE = "medium"
-DEFAULT_POSITION = "bottom-right"
-DEFAULT_TRANSPARENCY = "default"
-DEFAULT_COLOR = "grey"
-DEFAULT_INTERRUPT = False
-DEFAULT_TIMEOUT = 5
-DEFAULT_ICON = (
-    "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR4nGP6zwAAAgcBApo"
-    "cMXEAAAAASUVORK5CYII="
+from .const import (
+    ATTR_COLOR,
+    ATTR_DURATION,
+    ATTR_FILE,
+    ATTR_FILE_AUTH,
+    ATTR_FILE_AUTH_DIGEST,
+    ATTR_FILE_PASSWORD,
+    ATTR_FILE_PATH,
+    ATTR_FILE_URL,
+    ATTR_FILE_USERNAME,
+    ATTR_FONTSIZE,
+    ATTR_INTERRUPT,
+    ATTR_POSITION,
+    ATTR_TRANSPARENCY,
+    CONF_COLOR,
+    CONF_DURATION,
+    CONF_FONTSIZE,
+    CONF_INTERRUPT,
+    CONF_POSITION,
+    CONF_TRANSPARENCY,
+    DEFAULT_TIMEOUT,
 )
 
-ATTR_DURATION = "duration"
-ATTR_FONTSIZE = "fontsize"
-ATTR_POSITION = "position"
-ATTR_TRANSPARENCY = "transparency"
-ATTR_COLOR = "color"
-ATTR_BKGCOLOR = "bkgcolor"
-ATTR_INTERRUPT = "interrupt"
-ATTR_IMAGE = "filename2"
-ATTR_FILE = "file"
-# Attributes contained in file
-ATTR_FILE_URL = "url"
-ATTR_FILE_PATH = "path"
-ATTR_FILE_USERNAME = "username"
-ATTR_FILE_PASSWORD = "password"
-ATTR_FILE_AUTH = "auth"
-# Any other value or absence of 'auth' lead to basic authentication being used
-ATTR_FILE_AUTH_DIGEST = "digest"
-
-FONTSIZES = {"small": 1, "medium": 0, "large": 2, "max": 3}
-
-POSITIONS = {
-    "bottom-right": 0,
-    "bottom-left": 1,
-    "top-right": 2,
-    "top-left": 3,
-    "center": 4,
-}
-
-TRANSPARENCIES = {
-    "default": 0,
-    f"0{PERCENTAGE}": 1,
-    f"25{PERCENTAGE}": 2,
-    f"50{PERCENTAGE}": 3,
-    f"75{PERCENTAGE}": 4,
-    f"100{PERCENTAGE}": 5,
-}
-
-COLORS = {
-    "grey": "#607d8b",
-    "black": "#000000",
-    "indigo": "#303F9F",
-    "green": "#4CAF50",
-    "red": "#F44336",
-    "cyan": "#00BCD4",
-    "teal": "#009688",
-    "amber": "#FFC107",
-    "pink": "#E91E63",
-}
+_LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_DURATION, default=DEFAULT_DURATION): vol.Coerce(int),
-        vol.Optional(CONF_FONTSIZE, default=DEFAULT_FONTSIZE): vol.In(FONTSIZES.keys()),
-        vol.Optional(CONF_POSITION, default=DEFAULT_POSITION): vol.In(POSITIONS.keys()),
-        vol.Optional(CONF_TRANSPARENCY, default=DEFAULT_TRANSPARENCY): vol.In(
-            TRANSPARENCIES.keys()
+# Deprecated in Home Assistant 2021.8
+PLATFORM_SCHEMA = cv.deprecated(
```

Comment:
This is incorrect use of `cv.deprecated`. It takes a key name for the deprecated config key. For platforms we normally don't use `cv.deprecated` but just log a warning during platform legacy setup.